ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/review/v1/review/clientlibs/site/js/reviewview.js

```diff
@@ -0,0 +1,239 @@
+     * distinguish between them. If a component wants to put a data-attribute X, the attribute in HTML would be
+     * data-{NS}-{IS}-x=""
+     * @type {string}
+     */
+    static IS = "adaptiveFormReview";
+    static bemBlock = 'cmp-adaptiveform-review';
+    static templateAttribute = 'data-cmp-review';
+    static DATA_ATTRIBUTE_VISIBLE = 'data-cmp-visible';
+    static selectors = {
+      self: "[data-" + this.NS + '-is="' + this.IS + '"]',
+      container: `.${Review.bemBlock}__container`,
+      templates: "template[data-cmp-review-fieldType]"
+    };
+    static intersectionOptions = {
+      root: null,
+      rootMargin: "0px",
+      threshold: Review.buildThresholdList(),
+    };
+    static buildThresholdList() {
+      let thresholds = [];
+      let numSteps = 20;
+      for (let i = 1.0; i <= numSteps; i++) {
+        thresholds.push(i / numSteps);
+      }
+      thresholds.push(0);
+      return thresholds;
+    }
+
+    constructor(params) {
+      super(params);
+      this.#attacheIntersectionObserver();
+      this.#attacheClickEvent();
+      this.setTemplateMapping();
+    }
+    getContainer() {
+      return this.element.querySelector(Review.selectors.container);
+    }
+    getTemplates() {
+      return this.element.querySelectorAll(Review.selectors.templates);
+    }
+    getIntersectionElement() {
+      return this.element.querySelector("[" + Review.templateAttribute + "-intersection=" + this.id + "]");
+    }
+    getWidget() { return null }
+    getDescription() { return null; }
+    getLabel() { return null; }
+    getErrorDiv() { return null; }
+    getTooltipDiv() { return null; }
+    getQuestionMarkDiv() { return null; }
+    setModel(model) {
+      super.setModel(model);
+    }
+
+    #attacheClickEvent() {
+      this.getContainer().addEventListener('click', this.#clickHandler.bind(this))
+    }
+    #attacheIntersectionObserver() {
+      let observer = new IntersectionObserver(this.intersectionObserverHandler.bind(this), Review.intersectionOptions);
+      observer.observe(this.getIntersectionElement());
+    }
+    #clickHandler(event) {
+      if (event?.target?.nodeName === 'BUTTON') {
+        const id = event.target.getAttribute(Review.templateAttribute + '-fieldId');
+        this.formContainer.setFocus(id);
+      }
+    }
+    setTemplateMapping() {
+      const templates = this.getTemplates();
+      let mappings = {};
+      templates.forEach(template => {
+        const type = template.getAttribute(Review.templateAttribute + '-fieldType');
+        mappings[type || 'default'] = template;
+      });
+      this.templateMappings = mappings;
+    }
+    hasChild(panel) {
+      return panel && panel.children && panel.children.length;
+    }
+    isRepeatable(item) {
+      return item.fieldType === 'panel' && item.type === 'array'
+    }
+    showEditButton(fieldType, editAction) {
+      if (fieldType === 'panel' && (editAction === 'panel' || editAction === 'both')) {
```

Comment:
this is one if with two or conditins

Reply:
`if (editAction === 'both' || (editAction === 'panel' && fieldType === 'panel') || (editAction === 'field' && fieldType != 'panel'))`